>From c701061d7b3ded60d1a308aa2b004301df4a7bed Mon Sep 17 00:00:00 2001
From: Steve Langasek <steve.langasek@ubuntu.com>
Date: Fri, 17 May 2013 20:37:39 -0500
Subject: [PATCH] Packages with upstart jobs should carry a versioned
 dependency on sysv-rc

Depend on the version of invoke-rc.d that supports upstart jobs directly,
without which invoke-rc.d will incorrectly use the sysvinit script instead
of the upstart job when starting the service.
---
 dh_installinit | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/dh_installinit b/dh_installinit
index e227006..3260866 100755
--- a/dh_installinit
+++ b/dh_installinit
@@ -224,6 +224,13 @@ foreach my $package (@{$dh{DOPACKAGES}}) {
 
 	my $init=pkgfile($package,$script) || pkgfile($package,"init") ||
 		pkgfile($package,"init.d");
+
+	if ($job ne '' || ($dh{ONLYSCRIPTS} && -e "$tmp/etc/init/$jobfile.conf")) {
+		# minimal version of invoke-rc.d that supports upstart jobs
+		# directly
+		addsubstvar($package, "misc:Depends", "sysv-rc (>= 2.88dsf-24)");
+	}
+
 	if ($init ne '' && ! $dh{ONLYSCRIPTS}) {
 		if (! -d "$tmp/etc/init.d") {
 			doit("install","-d","$tmp/etc/init.d");
-- 
1.8.1.2

